### PR TITLE
Controllers spec

### PR DESCRIPTION
**Pull request by binyamolango**
# Hi Reviewer 👋

Here are the changes I implemented to the project.

1. Added RSpec as a gem to my project.
2. Created a Request spec file for all my controllers.
3. Added test for each action: 

- If the response status was correct
- If a correct template was rendered.
- If the response body includes the correct placeholder text.

### CODE REVIEW

File: spec/requests/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :request do
+  describe 'GET /index' do
+    it 'returns http success for users#index action' do
+      get users_url
+      expect(response).to have_http_status(:success) # 200
+      expect(response.body).to include('Here is a list of users')
+    end
```
Comment (thecodechaser) [inline spec/requests/user_spec.rb]: - [ ] Please make sure to add the below test for the user index action as requested in the project requirements.

image

File: spec/requests/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    @user = User.create(name: 'Binyam', photo: 'url', bio: 'I am a software developer.')
+    @post = Post.create(author: @user, title: 'Hello', text: 'Good career choice.', comments_count: 3, likes_count: 4)
+  end
+
+  describe 'GET /index' do
+    it 'returns http success for posts#index action' do
+      get user_posts_url(user_id: @user.id)
+      expect(response).to have_http_status(:success) # 200
+      expect(response.body).to include('Here is a list of posts for a given user')
+    end
```
Comment (thecodechaser) [inline spec/requests/post_spec.rb]: - [ ] Please make sure to add the below test for the post index action as requested in the project requirements.

image

File: db/schema.rb [hunk 3]
```diff
@@ -1,69 +1,69 @@
...
+  enable_extension "plpgsql"
+
+  create_table "comments", force: :cascade do |t|
+    t.text "text"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.string "address"
+    t.bigint "user_id", null: false
+    t.bigint "post_id", null: false
+    t.index ["address"], name: "index_comments_on_address"
```
Comment (thecodechaser) [inline db/schema.rb:24]: - [ ] I was not able to check if the tests are passing. While migration I'm getting the below errors. Please make sure to fix this and all tests should pass without any error.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @binyamolango👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job with controller specs. ✔️
- No linter errors. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @binyamolango,

## APPROVED 🟢

## Highlights
✔️ Passing tests

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by binyamolango** [comment]
@DammyShittu Thanks man!

**Pull request merged**